### ISSUE TITLE
Issue with "ARIA: row role": rowindex even necessary with all rows in the DOM

### ISSUE THREAD

**Issue by andypillip** [mdn/content#8389]
MDN URL: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Row_Role

#### What information was incorrect, unhelpful, or incomplete?

I'm not sure about the statement "If all the rows are present in the DOM, this attribute is not necessary." concerning `aria-rowindex`.

#### Specific section or headline?

* aria-rowindex attribute
* aria-colindex attribute

#### What did you expect to see?

Rather than referring to the DOM, we could refer to the accessibility tree.

Even rows with `display: none` will be hidden from screen readers and skipped in the row count. Hence, changing to `display: initial` would result in row indices to change and to confusing results for screen reader users.

#### Did you test this? If so, how?

I tested with a simple table that had one row with `display: none`, in Firefox 92.0b7 and NVDA 2020.4. Orca does not read the row number at all, it seems.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/accessibility/aria/roles/row_role`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Row_Role
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/accessibility/aria/roles/row_role/index.html
* Last commit: https://github.com/mdn/content/commit/1e8c048d86584064ca418a214063cb965fab3e71
* Document last modified: 2021-03-23T09:11:33.000Z

</details>

**Comment by sideshowbarker** [comment]
@scottaohara and @stevefaulkner may have some insight

**Comment by scottaohara** [comment]
This is correct per the aria spec. These attributes are best used for the virtualization of data, as if all the rows/cols are in the dom, browsers can calculate the number. Setting a row to display none and not having it be part of the current row count is expected behavior. 

The working group actually discussed these attributes last week, and there should be some editorial work done to hopefully clarify this topic in the spec.

**Comment by sideshowbarker** [comment]
Thanks @andypillip for raising this, and thanks @scottaohara for clarifying.

I’m going ahead and closing this since per https://github.com/mdn/content/issues/8389#issuecomment-908245640 it seems we don’t need a change to the content here. But if I’ve misunderstood, this can be revisited.

**Comment by andypillip** [comment]
Thanks everybody for your replies.

I'm still not sure though. The doc is intended for web authors, right?

A browser could calculate row numbers based on DOM despite hidden rows, but they don't. So for web author's practical work, it's actually the same: whether rows are hidden, or not in the DOM, in both cases they would need to apply the attributes.

Maybe I'll understand better once the section got clarified.
